library/tinkerpop/src/main/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopGraph.java

```diff
@@ -551,12 +554,38 @@ public Iterator<Edge> edges(final Object... elementIds) {
                 .then(new Limit<>(getAllElementsLimit, true))
                 .build();
         } else {
-            getOperation = new GetElements.Builder()
-                    .input(getElementSeeds(Arrays.asList(elementIds)))
-                    .view(new View.Builder()
+            // Partition elementIds by String[] or Object
+            // The assumption is that the String[] items are edge labels
+            // e.g. elementIds = {"1", ['knows']}
```

Comment:
I don't know if this assumption is quite right, edge IDs will usually always be in the form of `"[source, dest]"` this would show as a plain string not a string array as the gremlin query would be something like `g.E("[source, dest]")` or `g.E(["[source1, dest1]","[source2, dest2]"])`. Wouldn't if it was a single edge ID it be mistaken for a element seed? I think you might be confusing IDs with labels here?

Reply:
Yeah I wasn't sure about this either ... but when I debug it the params passed are `{"1", ["knows"]}`
Easiest example is the federated query that I changed that used `.outE('knows')`
Unless the root of the whole problem is that the params passed are incorrect?